src/extension/src/azure/azure-auth/azureAuth.ts

```diff
@@ -125,14 +125,20 @@ export abstract class AzureAuth {
         subscriptionClient.subscriptions,
         subscriptionClient.subscriptions.list()
       );
-      subscriptionItems.push(
-        ...subscriptions.map(subscription => ({
-          label: subscription.displayName || "",
-          subscriptionId: subscription.subscriptionId || "",
-          session,
-          subscription
-        }))
-      );
+
+      let filteredSubscriptions = subscriptions.reduce((items, subscription) => {
+        if (subscription.subscriptionId && !(subscriptionItems.some(item => item.subscriptionId === subscription.subscriptionId))) {
```

Comment:
NIT: This seems a bit inefficient (`O(n^2)` where we could do `O(n)` by filtering the list in one pass). I understand that most people will have very few subscriptions though. Also, where are the duplicate subscriptions coming from? Is there a way to query for unique list of subscriptions instead?

Reply:
Makes sense.

What do you think of the following approach?
```TypeScript
let subscriptionItems: SubscriptionItem[] = [];
let subscriptionIds: Map<string, boolean> = new Map();
for (const session of this.api.sessions) {
    // ..
    for (const subscription of subscriptions) {
        if ( subscriptionIds.has( subscription.subscriptionId ) ) {
            continue;
        }
        subscriptionIds.set( subscription.subscriptionId, true );
        subscriptionItems.push( ... )
    }
}
```

Reply:
In your example the subscriptionId check is missing. 

How about checking against the subscriptionItems array instead of subscriptionIds map?

I also remove the continue statement.

```javascript
let subscriptionItems: SubscriptionItem[] = [];
for (const session of this.api.sessions) {
    // ..
    for (const subscription of subscriptions) {
        if (subscription.subscriptionId && !subscriptionItems.some(item => item.subscriptionId === subscription.subscriptionId )) {
            subscriptionItems.push( ... )
        }        
    }
}
```
What do you think about this approach?